pyvista/plotting/scalar_bars.py

```diff
@@ -35,16 +35,23 @@ def __repr__(self):
         return '\n'.join(lines)
 
     def _remove_mapper_from_plotter(self, actor, reset_camera=False, render=False):
-        """Remove an actor's mapper from the given plotter's _scalar_bar_mappers."""
+        """Remove an actor's mapper from the given plotter's _scalar_bar_mappers.
+
+        This ensures that when actors are removed, their corresponding
+        scalar bars are removed.
+
+        """
         try:
             mapper = actor.GetMapper()
         except AttributeError:
             return
-        for name in list(self._scalar_bar_mappers):
+
+        for name in self._scalar_bar_mappers:
             try:
```

Comment:
Not sure if you remember, but we talked about this in https://github.com/pyvista/pyvista/pull/1347#pullrequestreview-674843704 and surrounding discussion. Just making sure that this change is intentional in light of new information.

Reply:
See #1554